aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,8 +1,315 @@
-.. include:: /../common/stub-topic.txt
+Bundling and Minification
+=========================
+
+By `Erik Reitan`_ 
+
+Bundling and minification are two techniques you can use in ASP.NET 5 to improve page load performance for your web application. Bundling makes it easy to combine or bundle multiple files into a single file. Minification performs a variety of different code optimizations to scripts and CSS, which results in smaller file sizes.
+
+This article explains and shows how to use the bundling and minification features included with the ASP.NET 5 Web Application template in `Visual Studio 2015 <http://go.microsoft.com/fwlink/?LinkId=517106>`_. 
+
+Overview
+--------
+Because bundling combines multiple files into a single file, it reduces the number of requests to the server that is required to retrieve and display a web asset, such as a web page. You can create CSS, JavaScript and other bundles. Fewer files, means fewer HTTP requests from your browser to the server or from the service providing your application. This results in improved first page load performance.
+
+Minification performs a variety of different code optimizations to reduce the size of requested assets (such as CSS, image, JavaScript files). Common results of minification include removing unnecessary white space and comments, and shortening variable names to one character. 
+
+Consider the following JavaScript function:
+
+.. code-block:: javascript
+
+	AddAltToImg = function (imageTagAndImageID, imageContext) {
+		///<signature>
+		///<summary> Adds an alt tab to the image
+		// </summary>
+		//<param name="imgElement" type="String">The image selector.</param>
+		//<param name="ContextForImage" type="String">The image context.</param>
+		///</signature>
+		var imageElement = $(imageTagAndImageID, imageContext);
+		imageElement.attr('alt', imageElement.attr('id').replace(/ID/, ''));
+	}
+
+After minification, the function is reduced to the following:
+
+.. code-block:: javascript
+
+	AddAltToImg = function (n, t) { var i = $(n, t); i.attr("alt", i.attr("id").replace(/ID/, "")) }
+	
+In addition to removing the comments and unnecessary whitespace, the following parameters and variable names were renamed (shortened) as follows:
+
+==================  =======  
+Original            Renamed      
+==================  =======  
+imageTagAndImageID  n  
+imageContext        t  
+imageElement        i   
+==================  =======  
+
+ASP.NET Web Application Template 
+--------------------------------
+The ASP.NET 5 Web Application project template is used to help you get started designing and coding a new Web application. It contains default functionality that demonstrates many aspects of ASP.NET. The template also includes the Node Package Manager (`npm <https://www.npmjs.com/>`_) and gulp by default.
+
```

Comment:
Node Package Manager (NPM) - so when you use npm below they will make the connection.